### PR TITLE
fix rust inlay hints when label is hashtable instead of string

### LINKED ISSUE #3896: With recent releases of rust-analyzer only few inlay hints show up
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [X] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

lsp-mode version: `20230106.1721`

With `lsp-rust-analyzer-server-display-inlay-hints` set to `t` I get the following with rust-analyzer release 2022-12-19:
image

With rust-analyzer release 2022-12-26:
image

I didn't see any errors in the lsp io log, a few hints (types in for loops) do show up.

### Steps to reproduce

Open a rust buffer with `lsp-rust-analyzer-server-display-inlay-hints` enabled.

### Expected behavior

Inlay hints for variables etc. are shown.

### Which Language Server did you use?

rust-analyzer

### OS

Linux

### Error callstack

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by brotzeit**
close #3896

### CODE REVIEW

**Review by jcs090218 — APPROVED** [review]

File: clients/lsp-rust.el [hunk 1]
```diff
@@ -1339,14 +1339,15 @@ meaning."
...
                    (overlay (make-overlay pos pos nil 'front-advance 'end-advance)))
-             (when (stringp label)
-               (overlay-put overlay 'lsp-rust-analyzer-inlay-hint t)
-               (overlay-put overlay 'before-string
-                            (format "%s%s%s"
-                                    (if padding-left " " "")
-                                    (propertize (lsp-rust-analyzer-format-inlay label kind)
-                                                'font-lock-face (lsp-rust-analyzer-face-for-inlay kind))
-                                    (if padding-right " " "")))))))
+             (when (hash-table-p label)
```
Comment (yyoncho) [inline clients/lsp-rust.el:1342]: Use `lsp-structure-p`

**Comment by npajkovsky** [comment]
Pitty, I cannot write anything in the lisp. It's pretty obvious what's wrong.

```
 "label": [
      {
        "value": ": "
      },
      {
        "value": "File",
        "location": {
          "uri": "file:///home/nikola/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/fs.rs",
          "range": {
            "start": {
              "line": 97,
              "character": 11
            },
            "end": {
              "line": 97,
              "character": 15
            }
          }
        }
      },
      {
        "value": ""
      }
```

**Comment by npajkovsky** [comment]
NAK to this MR. I have a working solution, and I'll submit it tomorrow. My version works with the latest rust-analyzer.

**Comment by brotzeit** [comment]
I missed the comment about `lsp-map`, sorry. But it probably doesn't make a big difference.

**Pull request closed without merging**